### ISSUE TITLE
class and style binding enhancements

### ISSUE THREAD

**Issue by jdanyow** [aurelia/binding#294]
investigate these scenarios:

``` html
<div class.my-class.bind="foo"
     style.background-color.bind="backgroundColor">
</div>
```

and:

``` html
<div class.bind="{ 'my-class': foo }"
     style.bind="{ 'background-color': backgroundColor }">
</div>
```

related to https://github.com/aurelia/binding/pull/293

**Comment by z-brooks** [comment]
Also related: https://github.com/aurelia/templating-binding/issues/23
A while ago I cobbled together a [plugin](https://github.com/z-brooks/aurelia-class-attribute) to use `class.bind=""` in my project. so, will this be in the parser as mentioned in #293, or a resource I can disable if I need mine?

**Comment by corneliutusnea** [comment]
This would be great as currently writing a component with lots of properties (e.g. a grid) makes the code look like a mess as I have to declare every property possible on the main grid instead of being able to declare them on sub-objects.
It would be so good to be able to bind to sub-properties:
```html
<grid options.selection.type="multiselection" options.selection.mode="checkboxes" ...
```

**Comment by EisenbergEffect** [comment]
I don't think we're trying to make a general solution here. Instead we are working on making styles better. If you have sub-objects used for configuration, you may want to consider using child elements in html.

**Comment by Alexander-Taran** [comment]
Stale since january 2016

**Comment by EisenbergEffect** [comment]
I think we should keep this open. Some of this can be done more easily in vNext potentially.

**Comment by fkleuver** [comment]
I thought about making a proper style declaration parser + AST as a plugin for vNext. That should make it a breeze to solve this issue as well as https://github.com/aurelia/binding/issues/680
@EisenbergEffect shall we rephrase this as a todo for vNext or what do you think?

**Comment by Alexander-Taran** [comment]
@EisenbergEffect added to vNext candidates list.
Can be closed
